streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/DruidEventMapper.java

```diff
@@ -55,20 +57,27 @@ public DruidEvent(StreamlineEvent event) {
             this.event = event;
         }
 
-        public static boolean isMap(Object map) {
-            return map instanceof Map;
-        }
-
         @Override
         protected Map<String, Object> delegate() {
```

Comment:
the `delegate` is going to be invoked multiple times, once per map operation, so maybe you can save the flattened result once and re-use it.